### ISSUE TITLE
zfs-core-sys does not support illumos

### ISSUE THREAD

**Issue by Toasterson** [Inner-Heaven/libzetta-rs#184]
**Describe the bug**
When trying to compile this library with this simple test program 
```main.rs
use libzetta::zpool::open3::ZpoolOpen3;
use libzetta::zpool::ZpoolEngine;
                                                                                                                                                                                                                   
fn main() {
    let pool_handle = ZpoolOpen3::default();
    println!("{:#?}", pool_handle.exists("rpool"));
    println!("{:#?}", pool_handle.exists("zroot"));
}
```

I get the follwoing error on illumos
```
   Compiling zfs-core-sys v0.5.0
error: failed to run custom build command for `zfs-core-sys v0.5.0`
                                                                                                                                                                                                                   
Caused by:
  process didn't exit successfully: `/ws/toasty/ztest/target/debug/build/zfs-core-sys-a0c58ea6900454a9/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=LIBZFS_CORE_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-illumos
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_illumos
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=LIBZFS_CORE_STATIC
  cargo:rerun-if-env-changed=LIBZFS_CORE_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-illumos
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_illumos
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-illumos
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_illumos
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-illumos
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_illumos
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
                                                                                                                                                                                                                   
  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: `PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" PKG_CONFIG_ALLOW_SYSTEM_LIBS="1" "pkg-config" "--libs" "--cflags" "libzfs_core"` did not exit successfully: exit status: 1
  error: could not find system library 'libzfs_core' required by the 'zfs-core-sys' crate
                                                                                                                                                                                                                   
  --- stderr
  Package libzfs_core was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libzfs_core.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'libzfs_core' found
  ', /ws/toasty/.cargo/registry/src/github.com-1ecc6299db9ec823/zfs-core-sys-0.5.0/build.rs:7:50
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**To Reproduce**
- Make a new binary crate with `cargo new` on illumos
- Add this library as dependency
- use the library (see sample code above)

**Expected behavior**
```
  Downloaded derive_builder_macro v0.12.0
  Downloaded derive_builder_core v0.12.0
  Downloaded derive_builder v0.12.0
  Downloaded 3 crates (73.2 KB) in 1.75s
   Compiling zfs-core-sys v0.5.0
   Compiling derive_builder_core v0.12.0
   Compiling derive_builder_macro v0.12.0
   Compiling derive_builder v0.12.0
   Compiling libzetta v0.3.1 (https://github.com/Solarm-Project/libzetta-rs#82c38d5b)
   Compiling ztest v0.1.0 (/ws/toasty/ztest)
    Finished dev [unoptimized + debuginfo] target(s) in 11.90s
     Running `target/debug/ztest`
Ok(
    true,
)
Ok(
    false,
)
```


** Please complete the following information:**
 - OS: OpenIndiana Hipster (illumos)
 - libZetta 0.3.1

**Additional context**
The current version of zfs-core-sys has code in build.rs that handles this case but no release has been made yet https://github.com/jmesmon/rust-libzfs/blob/master/zfs-core-sys/build.rs Trying to update the dependency gets me into a dependency hell between nvpair-sys and libnv so I think you may need to start forking the sys crates to continue in the project.

I could get the program to work by faking the pkg-config file with this 
```
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include
                                                                                                                                                                                                                   
Name: libzfs_core
Description: LibZFS core library
Version: 2.1.9
URL: https://github.com/openzfs/zfs
Requires.private:    zlib
Cflags: -I${includedir}
Libs: -L${libdir} -lzfs_core -lnvpair
Libs.private:  -L/usr/lib -lm -pthread
```

and adding the current path to the pkg-config search path `PKG_CONFIG_PATH_x86_64_unknown_illumos="/usr/lib/pkgconfig:$(realpath .)" cargo run`

**Comment by andoriyu** [comment]
@Toasterson thanks for the issue. I guess forking sys crate is the way, since it's been 10 months since the change was introduced and no release has been cut. 

I will try to find time to work on this and upgrade libnv dependency.

Do you have any issues with https://github.com/Inner-Heaven/libnv-rs building for you?

**Comment by Toasterson** [comment]
No it compiled and worked. Only thing that happened was the dependency hell where libnv couldn't update nvpair-sys but no compile error.

**Comment by andoriyu** [comment]
Should be closed in `0.4.2`. Please check when you get a chance.
